packages/components/src/List/ListComposition/DisplayMode/DisplayMode.component.js

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { translate } from 'react-i18next';
+import { Navbar, NavDropdown, Nav, MenuItem } from 'react-bootstrap';
+import uuid from 'uuid';
+
+import { ListContext } from '../context';
+import Icon from '../../../Icon';
+import getDefaultT from '../../../translate';
+import I18N_DOMAIN_COMPONENTS from '../../../constants';
+
+function getIcon(selected) {
+	switch (selected) {
+		case 'table':
+			return 'talend-table';
+		case 'large':
+			return 'talend-expanded';
+		default:
+			return 'talend-table';
+	}
+}
+
+class DisplayMode extends React.Component {
+	static displayName = 'List.DisplayMode';
+	static propTypes = {
+		displayModes: PropTypes.arrayOf(PropTypes.string),
+		id: PropTypes.string,
+		mode: PropTypes.string,
+		onChange: PropTypes.func.isRequired,
+		t: PropTypes.func,
+	};
+	static defaultProps = {
+		id: uuid.v4(),
+		displayModes: ['table', 'large'],
+		mode: 'table',
+		t: getDefaultT(),
+	};
+
+	getLabel(mode) {
+		const { t } = this.props;
+		switch (mode) {
+			case 'table':
+				return t('LIST_SELECT_DISPLAY_MODE_TABLE', { defaultValue: 'Table' });
+			case 'large':
+				return t('LIST_SELECT_DISPLAY_MODE_LARGE', { defaultValue: 'Expanded' });
+			default:
+				return t('LIST_SELECT_DISPLAY_MODE_TABLE', { defaultValue: 'Table' });
+		}
+	}
+
+	getMenuItems() {
+		return this.props.displayModes.map(option => {
+			const displayModeLabel = this.getLabel(option);
```

Comment:
Can't you keep a consistency between your variable name. Option is a display mode, you call it option and in getLabel is named mode.
One of my convention is Entities.map => entity

Reply:
`displayMode` is already the name used in context.